model/infinispan/src/main/java/org/keycloak/connections/infinispan/DefaultInfinispanConnectionProviderFactory.java

```diff
@@ -290,10 +291,16 @@ public Map<String, String> getOperationalInfo() {
         return info;
     }
 
+    @Override
+    public boolean isClusterHealthy() {
+        clusterHealth.triggerClusterHealCheck();
+        return clusterHealth.isHealthy();
```

Comment:
The default operator configuration for the StatefulSet uses a timeout of 10 seconds, so in this case it will be 20 seconds until the split-brain is detected by the readiness probe as the first call to this method will result in `true` being returned?

Reply:
It depends on the failure threshold. What you describe is the minimum time to mark it not ready.